omod/src/main/java/org/openmrs/web/rest/resource/TagResource.java

```diff
@@ -0,0 +1,265 @@
+	 */
+	@Override
+	protected void delete(Tag tag, String reason, RequestContext context) {
+		throw new ResourceDoesNotSupportOperationException("delete not allowed on tag");
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#purge(Object, RequestContext)
+	 */
+	@Override
+	public void purge(Tag tag, RequestContext context) {
+		getService().purgeTag(tag);
+	}
+	
+	/**
+	 * Returns the DefaultRepresentation
+	 */
+	@RepHandler(DefaultRepresentation.class)
+	public SimpleObject asDef(Tag delegate) throws ConversionException, ClassNotFoundException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("tag");
+		if (StringUtils.isNotBlank(delegate.getObjectType())) {
+			description.addProperty("uuid");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+		}
+		List<Tag> tagList = Context.getService(TagService.class).getTags(delegate.getTag(), true);
+		Iterator<Tag> tagIterator = tagList.iterator();
+		while (tagIterator.hasNext()) {
+			Tag iterator = tagIterator.next();
+			description.addLink(getResourceName(iterator.getObjectType()),
+			    getUriOfObject(iterator.getObjectType(), iterator.getObjectUuid()));
+		}
+		description.addSelfLink();
+		description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		if (tagList.size() == 0) {
+			description = null;
+		}
+		return convertDelegateToRepresentation(delegate, description);
+	}
+	
+	@RepHandler(FullRepresentation.class)
+	public SimpleObject asFull(Tag delegate) throws ConversionException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		if (StringUtils.isNotBlank(delegate.getObjectType())) {
+			description.addProperty("uuid");
+			description.addProperty("tag");
+			description.addProperty("objectType");
+			description.addProperty("objectUuid");
+			description.addProperty("auditInfo");
+		} else {
+			List<Tag> tags = getService().getTags(delegate.getTag(), true);
+			SimpleObject simpleObject = new SimpleObject();
+			List<Map<String, Object>> taggedObjects = new ArrayList<Map<String, Object>>();
+			for (Tag t : tags) {
+				Map<String, Object> taggedObject = new LinkedHashMap<String, Object>();
+				taggedObject.put("tag", t.getTag());
+				taggedObject.put("uuid", t.getUuid());
+				taggedObject.put("objectType", t.getObjectType());
+				taggedObject.put("objectUuid", t.getObjectUuid());
+				taggedObjects.add(taggedObject);
+			}
+			simpleObject.put("taggedObjects", taggedObjects);
+			return simpleObject;
+		}
+		description.addSelfLink();
+		return convertDelegateToRepresentation(delegate, description);
+	}
+	
+	/**
+	 * @see BaseDelegatingResource#getRepresentationDescription(Representation)
+	 */
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = null;
+		return description;
```

Comment:
What is the point of declaring a variable only to always return null, just return null and that's it